src/Directory.Build.targets

```diff
@@ -7,7 +7,7 @@
   -->
   <Import Project="packaging.stubs.targets" Condition="'$(MSBuildProjectExtension)' != '.pkgproj'" />
```

Comment:
Why isn't this a "relative pattern"? Are you talking about `..` in particular?

This PR is striking me as extremely odd, since we rely on this working as-is in so much... why now?

Reply:
> Relative paths in imported projects are interpreted relative to the directory of the importing project. Therefore, if a project file is imported into several project files in different locations, the relative paths in the imported project file will be interpreted differently for each imported project.

from https://docs.microsoft.com/en-us/visualstudio/msbuild/import-element-msbuild?view=vs-2019#remarks.
This is what is causing troubles for installer.tasks. Using absolute paths prevents that. For the sake of fixing the runtime repo, the change could be scoped down to only the two Directory.Build.props/targets files under tools-local but I thought it might a good idea to harden this everywhere.

Reply:
For example, the simplest thing, `<Import Project="..\Directory.Build.props" />`, should not work if this PR is correct.

Reply:
That doc seems incorrect. If it were only referring to properties and items, then it would be correct by my experience.

Reply:
What part do you believe is incorrect?

Reply:
The term `relative paths` is too broad in

> Relative paths in imported projects are interpreted relative to the directory of the importing project.

It's true that paths in properties aren't evaluated at all, so their meaning is defined based on where they're used, and item includes are expanded in the context of the importing project. However, imports must be relative to the file they appear in, or else it's in contradiction with even the most basic dir.props chains in practice:

```
src/foo/foo.csproj
src/foo/Directory.Build.props
src/Directory.Build.props
Directory.Build.props
```

Each `Directory.Build.props` contains `<Import Project="..\Directory.Build.props" />`. When `src/Directory.Build.props` is imported, if the above doc is true, then the file would try to import itself.

Reply:
@rainersigwald can you help out here?

Reply:
An even simpler example is the line I left this comment on... `<Import Project="packaging.stubs.targets"` is against the doc, because `packaging.stubs.targets` exists in `src/`, not adjacent to every single project.

Reply:
I'm almost certain that `<Import>` paths are evaluated relative to the current file. In properties and items they're evaluated relative to the entry project file.

Reply:
Yeah, I think @dagood was right with https://github.com/dotnet/core-setup/pull/8785/files#r343228968. Since that condition isn't in the context of an import it gets "the usual" CWD rules, instead of the import-relative ones.

Reply:
I'll file the doc bug.

Reply:
Filed https://github.com/MicrosoftDocs/visualstudio-docs/issues/4287.

Reply:
Thanks Davis and Rainer. Any suggestion what to do with this? Should I remove changes in the Import statements or leave them to be absolute?

Reply:
I lean toward backing them out, but don't feel strongly about it.

Reply:
And thanks Jeremy as well of course.